### PR TITLE
EVG-15043 add attach/detach project mutation resolvers

### PR DESCRIPTION
**Pull request by ablack12**
[EVG-15043](https://jira.mongodb.org/browse/EVG-15043)

### Description 
Added resolvers for this; discussed the return value with Mohamed and learned about the cache 😇 

### Testing 
Added unit tests.

### CODE REVIEW

File: graphql/schema.graphql [hunk 1]
```diff
@@ -81,6 +81,8 @@ type Query {
 type Mutation {
   addFavoriteProject(identifier: String!): Project!
   removeFavoriteProject(identifier: String!): Project!
+  attachProjectToRepo(id: String!): Project!
+  detachProjectFromRepo(id: String!): Project!
```
Comment (khelif96) [inline graphql/schema.graphql:85]: attachUserToProject? replace id: with `projectId` ?

File: graphql/resolvers.go [hunk 2]
```diff
@@ -2008,6 +2008,46 @@ func (r *taskQueueItemResolver) Requester(ctx context.Context, obj *restModel.AP
...
+	usr := MustHaveUser(ctx)
+	pRef, err := r.sc.FindProjectById(id, false)
+	if err != nil {
+		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("error finding project %s: %s", id, err.Error()))
+	}
+	if pRef == nil {
+		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("cannot find project %s", id))
+	}
+	if err = pRef.DetachFromRepo(usr); err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("error attaching to repo: %s", err.Error()))
```
Comment (malikchaya2) [inline graphql/resolvers.go:2041]: *detaching from

**Review by malikchaya2 — APPROVED** [review]
Approved with minor comment about an error message

File: graphql/schema.graphql (hunk 1 above)
Comment (ablack12) [reply graphql/schema.graphql:85]: Are you suggesting a rename of attachProjectToRepo to attachUserToProject? That actually isn't what's happening in this function; we're turning on the option of defaulting to repo for the branch project.

**Review by khelif96 — APPROVED** [review]

File: graphql/schema.graphql (hunk 1 above)
Comment (khelif96) [reply graphql/schema.graphql]: oops didn't actually read what `DetachFromRepo` and `AttachFromRepo` did. I assumed they did something with the user that was passed into them, to give the user some permission.

**Pull request merged**